src/core/border_router/routing_manager.cpp

```diff
@@ -2237,20 +2243,48 @@ void RoutingManager::OnMeshPrefixArray::MarkAsDeleted(const OnMeshPrefix &aPrefi
 
 RoutingManager::Nat64PrefixManager::Nat64PrefixManager(Instance &aInstance)
     : InstanceLocator(aInstance)
+    , mEnabled(false)
     , mTimer(aInstance, HandleTimer)
 {
     mInfraIfPrefix.Clear();
     mLocalPrefix.Clear();
     mPublishedPrefix.Clear();
 }
 
+void RoutingManager::Nat64PrefixManager::SetEnabled(bool aEnabled)
+{
+    VerifyOrExit(mEnabled != aEnabled);
+
```

Comment:
I think we need `mEnabled = aEnabled` here?

Reply:
Thanks.